cl/api/views.py

```diff
@@ -164,62 +164,61 @@ def coverage_data(request, version, court):
     )
 
 
-def fetch_start_end_dates_for_court(court_id: str, court_name: str):
-    """Fetch start and end dates for court
+def fetch_start_end_dates_for_court(court_id: str):
```

Comment:
This function is a lot better, but you're still hiding your purpose and it's muddying your code below. 

What you're doing now is saying:

 - Give me the start and end dates for a court.

And then your function is like:

 - Here ya go — this complicated object has what you need!
 - (Sneaky, sneaky, I put some extra data in there too!)

And it sort of looks like you're avoiding a query this way, but you're not. You're querying all Clusters, and then joining the Court info to it (via the massive docket table), which is actually quite a bit worse, because you have to:

 - Filter to all clusters by court id
 - Join those clusters to their dockets
 - Join those dockets to their courts
 - Get the jurisdiction from the court table

It would be better if this function took a court ID, and then delivered on its contract by giving back two dates — nothing more sneaking through, nothing less.

Then, down below, where you need the jurisdiction for the court, just look it up in the *very* small court table. That will:

1. Be practically instant
2. Your code will deliver on the contracts implied by the function names
3. You won't have to do the [0] and [1] stuff below, making the code more clear

Reply:
I figured you were going to approve but I was ... as you know trying to save on queries.

Reply:
This has been updated.